Mathlib/Data/List/Infix.lean

```diff
@@ -507,4 +540,17 @@ theorem mem_of_mem_suffix (hx : a ∈ l₁) (hl : l₁ <:+ l₂) : a ∈ l₂ :=
   hl.subset hx
 #align list.mem_of_mem_suffix List.mem_of_mem_suffix
 
+theorem ne_nil_prefix {x y : List α} (hx : x ≠ []) (h : x <+: y) : y ≠ [] := by
```

Comment:
```suggestion
theorem IsPrefix.ne_nil {x y : List α} (hx : x ≠ []) (h : x <+: y) : y ≠ [] := by
```
to enable dot notation.

Reply:
Intuitively I'd prefer the current order, maybe just because in applications I used `apply ne_nil_prefix hx`. Moreover, dot notation would allow either `apply IsPrefix.ne_nil hx` or `apply h.ne_nil` with the current order, but only the second one (without _) with the suggested one. (Of course, there may be more relevant considerations)

Reply:
Dot notation doesn't care about hypothesis order (to the extent that you are not swapping two `IsPrefix` assumptions)

Reply:
I don't get it. Since dot notation doesn't care about hypothesis order, you get `h.ne_nil : x ≠ [] → y ≠ []` with either order, don't you?

Reply:
It is less confusing if `h.ne_nil h2` is the same as `IsPrefix.ne_nil h h2` instead of `IsPrefix.ne_nil h2 h`. I also prefer the order `x <+: y → x ≠ [] → y ≠ []`, but I don't feel strongly.